rfcs/text/0428-type-inheritance-and-forking.md

```diff
@@ -0,0 +1,952 @@
+A [versioned URI](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0408-versioning-types.md#type-uris) is used so that subtypes aren't automatically updated (and potentially invalidated) when the supertype is updated.
+
+As extended types can extend other extended types we must also make sure that there are no cycles within the type hierarchy, as it makes types difficult to resolve/reason about and could lead to unpredictable behavior.
+
+## Defining entity type duplication
+
+When looking for public types, if extending an entity type is insufficient, an alternative is to duplicate the type so that individual properties can be removed or overridden. Duplicated types become new complete, standalone types.
+
+Type duplication, unlike extended types, does not imply that instances of the original type can be substituted by instances of the duplicated type because the types may be incompatible. A mapping may be used to persist some of the compatibility, but mapping is yet to be defined within the type system.
+
+### Duplication strategy
+
+As outlined in the beginning, duplication can take place in many different ways for entity types. Since entity types can extend other entity types in this proposal, duplication can happen at different conceptual locations. As an example, if a user wanted to redefine an `Employee` that has the following type hierarchy:
+
+```txt
+name◄─────Being───┐
+                  │
+          age◄────Person─────Employee
+                             │
+   occupation◄───────────────┘
+```
+
+where `Employee` is a subtype of `Person` which in turn is a subtype of `Being`. The three properties `name`, `age` and `occupation` would be present on `Employee` and considered all expanded properties for `Employee`.
+The user wants to change the `age` property to a `tenure` property through duplication. They would have to create a new subtype of `Being` with the desired changes:
+
+```txt
+name◄─────Being───┐
+                  │
+       tenure◄────MyEmployee
+                  │
+   occupation◄────┘
+```
+
+From the perspective of the user duplicating a type, the `Person` and `Employee` entity types were squashed together or "expanded" for duplication to have the desired effect. The `Being` supertype relation is still kept when changing a property on the `MyEmployee` duplicated type. But in this process, the `MyEmployee` entity type loses the ability to trivially be treated as a `Person`, as there is no subtyping relation with `Person` on the new `MyEmployee` duplicated type.
+
+The duplicatio nhappened on the expanded type resulting in combining/expanding `Person` and `Employee`, but it would also have been possible expand the entire type, getting rid of `Being`. This would have resulted in the following type hierarchy:
```

Comment:
```suggestion
The duplication happened on the expanded type resulting in combining/expanding `Person` and `Employee`, but it would also have been possible to expand the entire type, getting rid of `Being`. This would have resulted in the following type hierarchy:
```